src/main/scala/dpla/ingestion3/harvesters/file/VaFileHarvester.scala

```diff
@@ -0,0 +1,132 @@
+package dpla.ingestion3.harvesters.file
+
+import java.io.{File, FileInputStream}
+import java.util.zip.ZipInputStream
+
+import dpla.ingestion3.confs.i3Conf
+import dpla.ingestion3.mappers.utils.XmlExtractor
+import org.apache.commons.io.IOUtils
+import org.apache.log4j.Logger
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import com.databricks.spark.avro._
+
+import scala.util.{Failure, Success, Try}
+import scala.xml.{MinimizeMode, Node, Utility, XML}
+
+
+/**
+  * Extracts values from parsed Xml
+  */
+class VaFileExtractor extends XmlExtractor
+
+/**
+  * Entry for performing a Digital Virginias file harvest
+  */
+class VaFileHarvester(spark: SparkSession,
+                      shortName: String,
+                      conf: i3Conf,
+                      logger: Logger)
+  extends FileHarvester(spark, shortName, conf, logger) {
+
+  def mimeType: String = "application_xml"
+
+  protected val extractor = new VaFileExtractor()
+
+  /**
+    * Loads .zip files
+    *
+    * @param file File to parse
+    * @return TarInputstream of the tar contents
+    */
+  def getInputStream(file: File): Option[ZipInputStream] = {
```

Comment:
Because we're only handling zips in this class, and they should already be filtered by the FilenameFilter above, I wonder if you even need the match statement here.

Reply:
I added a comment with this question so we don't lose it.  For now I'll leave it as is so its consistent with the other file harvesters.